sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_project_setup.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "3. Repository Setup"
+subtitle: Configuring your nf-core pipeline repository for testing with nf-test
+weight: 30
+---
+
+## Example nf-core Repository Structure
+
+Here is an example of an nf-core pipeline project with nf-test:
+
+```
+my-pipeline/
+├── main.nf                          # Main pipeline script
+├── nextflow.config                  # Main pipeline configuration
+├── nextflow_schema.json            # Pipeline parameter schema
+├── nf-test.config                  # nf-test configuration
+├── modules.json                    # Module dependencies
+├── tests/                          # Pipeline-level tests
+│   ├── default.nf.test            # Main pipeline test
+│   ├── nextflow.config            # Test-specific config
+│   └── .nftignore                 # Files to ignore in snapshots
+├── conf/                           # Configuration profiles
+│   ├── test.config                # Test profile
+│   ├── test_full.config           # Full test profile
+│   ├── base.config                # Base configuration
+│   └── igenomes.config            # Reference genome configurations
+├── assets/                         # Pipeline assets
+│   ├── samplesheet.csv            # Test samplesheet
+│   └── schema_input.json          # Input validation schema
+├── modules/nf-core/               # Module tests
+│   └── */tests/main.nf.test      # Individual module tests
+├── subworkflows/nf-core/          # Subworkflow tests
+│   └── */tests/main.nf.test      # Individual subworkflow tests
+└── .nf-test/                      # nf-test working directory
+```
+
+### 1. Key Configuration Files
+
+#### `nf-test.config` (Main Configuration)
+
+**Purpose**: Controls nf-test global settings
+
+```groovy
+config {
+    testsDir "."
+    workDir System.getenv("NFT_WORKDIR") ?: ".nf-test"
+    configFile "tests/nextflow.config"
+    ignore 'modules/nf-core/**/*', 'subworkflows/nf-core/**/*'
+    profile "test"
+    triggers 'nextflow.config', 'nf-test.config', 'conf/test.config', 'tests/nextflow.config', 'tests/.nftignore'
+    plugins {
+        load "nft-utils@0.0.3"
+        // Add plugins based on your pipeline needs:
+        // load "nft-bam@0.6.0"    // For BAM file testing
+        // load "nft-vcf@1.0.7"    // For VCF file testing
+        // load "nft-fastq@0.0.1"  // For FASTQ file testing
+    }
+}
+```
+
+> **For complete configuration options**, see the [official nf-test configuration documentation](https://www.nf-test.com/docs/configuration/).
+
+### Available nf-test Plugins
+
+For nf-core pipeline testing, you may need these plugins:
```

Comment:
I would explain slightly more explicitly that these plugins come with addition functions to check the contents of files